src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -978,6 +978,114 @@ public async Task DisableTwoFactorProviderAsync(Organization organization, TwoFa
             await UpdateAsync(organization);
         }
 
+        private async Task<List<OrganizationUser>> InviteUsersAsync(Guid organizationId, Guid? invitingUserId, IEnumerable<(OrganizationUserInvite, string)> invites)
+        {
+            var organization = await GetOrgById(organizationId);
+            if (organization == null || invites.Any(i => i.Item1.Emails == null || i.Item2 == null))
```

Comment:
On the `NotFoundException`, is the intent (outside of the obvious org not found) to determine if any invites provided don't have emails OR is the intent that no invites provided have emails?

If the later is the intended check, this should be retooled to create a new `IEnumerable<T>` of those invites which have emails and an external ID, and then check that `!validInvites.Any()` for this exception condition, and then use that same one later to already have a filtered set to use for pulling invite types, etc.

Reply:
The way I was thinking about this is that the organization must exist and all invites must be "actionable" or we error. I don't want to allow partial imports, which would confuse the directory connector in its current implementation.

This interpretation is equivalent to your first.